mlrun/api/db/sqldb/db.py

```diff
@@ -950,60 +1028,39 @@ def update_feature_set(
         session,
         project,
         name,
-        feature_set_update: schemas.FeatureSetUpdate,
+        feature_set_update: dict,
         tag=None,
         uid=None,
+        additive=False,
     ):
-        # If object uid was given, use it. Else, find it by tag
-        if not uid:
-            tag = tag or "latest"
-            tag = self._query(
-                session, FeatureSet.Tag, project=project, name=tag, obj_name=name
-            ).one_or_none()
-            if not tag:
-                feature_set_uri = generate_object_uri(project, name, tag)
-                raise mlrun.errors.MLRunNotFoundError(
-                    f"Feature-set not found {feature_set_uri}"
-                )
-            query = self._query(session, FeatureSet, id=tag.obj_id)
-        else:
-            query = self._query(
-                session, FeatureSet, name=name, project=project, uid=uid
-            )
-
-        feature_set_record = query.one_or_none()
+        feature_set_record = self._get_feature_set(session, project, name, tag, uid)
         if not feature_set_record:
             feature_set_uri = generate_object_uri(project, name, tag)
             raise mlrun.errors.MLRunNotFoundError(
                 f"Feature-set not found {feature_set_uri}"
             )
 
-        data = feature_set_update.dict()
-
-        # TODO (future) - granular update of JSON fields.
-        status = data.pop("status", None)
-        if status:
-            feature_set_record.status = status
-            feature_set_record.state = status.get("status") or feature_set_record.state
-
-        features = data.pop("features", [])
-        if features:
-            self._update_feature_set_features_or_entities(
-                feature_set_record, features, is_entity=False, replace=True
-            )
-
-        entities = data.pop("entities", [])
-        if entities:
-            self._update_feature_set_features_or_entities(
-                feature_set_record, entities, is_entity=True, replace=True
-            )
-
-        update_labels(feature_set_record, data.pop("labels", {}))
-        feature_set_record.updated = datetime.now(timezone.utc)
-
-        self._upsert(session, feature_set_record)
+        feature_set_struct = feature_set_record.dict()
+        # using mergedeep for merging the patch content into the existing dictionary
+        strategy = Strategy.REPLACE
+        if additive:
+            strategy = Strategy.ADDITIVE
+        merge(feature_set_struct, feature_set_update, strategy=strategy)
 
-        return feature_set_record.id
+        versioned = not feature_set_struct["metadata"]["uid"].startswith(
```

Comment:
```suggestion
        versioned = not feature_set_record.uid.startswith(
```
doesn't matter but nicer IMO